src/main/java/tetris/Controller.java

```diff
@@ -32,24 +32,31 @@
         Platform.runLater(() -> Action.SOFT_DROP.attempt(tetromino, grid));
         Platform.runLater(() -> redraw());
     });
-
     private Settings settings;
+    private TetrominoFactory factory;
 
     /**
      * the Controller class determines the game flow and does the actual event
      * handling.
-     * 
+     *
      * @param ui
-     *            the application in which the game is running
+     * @param settings
+     * @param factory
+     * @param soundManager
+     * @param score
+     * @param scoreBoard
      */
-    public Controller(View ui, Settings settings) {
+    public Controller(View ui, Settings settings, TetrominoFactory factory,
```

Comment:
Except for one thing: an interface can have multiple implementations whereas this static factory can't. That is exactly what I'm trying to solve here. I would also like to remind you that it would make testing easier, because with Dependency Injection it would allow you to mock the Factory. The dropNewTetromino() method is currently not tested btw: https://codecov.io/gh/bstinenbosch/tetris/src/f76700bb6ad5163a120626d00ea9b64eabd8a4a7/src/main/java/tetris/Controller.java